library/Vanilla/Permissions.php

```diff
@@ -274,6 +293,21 @@ public function setBanned($isBanned) {
     }
 
     /**
+     * Get the flag to determine if all permission checks are being overridden.
+     *
+     * @param bool|null $overrideAll
+     * @return $this
+     */
+    public function setOverrideAll($overrideAll) {
+        if ($overrideAll !== null) {
+            $overrideAll = boolval($overrideAll);
```

Comment:
Down the way it encourage lazy behaviours and we get bugs like this: https://github.com/vanilla/vanilla/pull/5100#pullrequestreview-20389295